egs2/namine_ritsu_utagoe_db/svs1/local/dataset_split.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import os
+import shutil
+
+UTT_PREFIX = "namine"
+DEV_LIST = [
+    "subetenovocaloidnisensenfukoku",
+    "Choir",
+    "shumatsutokiseki",
+    "gunjoirono",
+    "dokomademomukuni",
+]
+TEST_LIST = ["hoshifuru", "hanayorikimini", "yume_shigure", "haruwasakura", "reverse"]
+
+
+def train_check(song):
+    return (song not in DEV_LIST) and (song not in TEST_LIST)
+
+
+def dev_check(song):
+    return song in DEV_LIST
+
+
+def test_check(song):
+    return song in TEST_LIST
+
+
+def pack_zero(string, size=20):
+    if len(string) < size:
+        string = "0" * (size - len(string)) + string
+    return string
+
+
+def makedir(data_url):
+    if os.path.exists(data_url):
+        shutil.rmtree(data_url)
+
+    os.makedirs(data_url)
+
+
+def process_text_info(text):
+    info = open(text, "r", encoding="utf-8")
+    label_info = []
+    text_info = []
+    for line in info.readlines():
+        line = line.strip().split()
+        label_info.append(
+            "{} {} {}".format(
+                float(line[0]) / 1e7, float(line[1]) / 1e7, line[2].strip()
+            )
+        )
+        text_info.append(line[2].strip())
+    return " ".join(label_info), " ".join(text_info)
+
+
+def process_subset(src_data, subset, check_func, fs, wav_dump):
+    src_data = os.path.join(src_data, "「波音リツ」歌声データベースVer2/DATABASE")
+    subfolder = os.listdir(src_data)
+    makedir(subset)
+    wavscp = open(os.path.join(subset, "wav.scp"), "w", encoding="utf-8")
+    utt2spk = open(os.path.join(subset, "utt2spk"), "w", encoding="utf-8")
+    # midiscp = open(os.path.join(subset, "midi.scp"), "w", encoding="utf-8")
+    # text_scp = open(os.path.join(subset, "text"), "w", encoding="utf-8")
+    label_scp = open(os.path.join(subset, "label"), "w", encoding="utf-8")
+    musicxmlscp = open(os.path.join(subset, "score.scp"), "w", encoding="utf-8")
+
+    for folder in subfolder:
+        if not os.path.isdir(os.path.join(src_data, folder)):
+            continue
+        if not check_func(folder):
+            continue
+        utt_id = "{}_{}".format(UTT_PREFIX, pack_zero(folder))
+
+        cmd = f"sox {os.path.join(src_data, folder, folder)}.wav -c 1 -t wavpcm -b 16 -r {fs} {os.path.join(wav_dump, folder)}_bits16.wav"
+        os.system(cmd)
+
+        wavscp.write(
+            "{} {}\n".format(
+                utt_id, os.path.join(wav_dump, "{}_bits16.wav".format(folder))
+            )
+        )
+        utt2spk.write("{} {}\n".format(utt_id, UTT_PREFIX))
+        label_info, text_info = process_text_info(
+            os.path.join(src_data, folder, "{}.lab".format(folder))
+        )
+        # text_scp.write("{} {}\n".format(utt_id, text_info))
```

Comment:
Why do you remove the write of text_scp and midiscp?

Reply:
> Why do you remove the write of text_scp and midiscp?

I think they are not used in this dataset. I find that ofuton also doesn't use text_scp and midiscp, so I removed them. Should I keep them?

Reply:
If it is not used, let's remove them.